repose-aggregator/components/filters/openstack-identity-v3/src/main/scala/org/openrepose/filters/openstackidentityv3/OpenStackIdentityV3Handler.scala

```diff
@@ -26,225 +26,149 @@ import com.rackspace.httpdelegation.HttpDelegationManager
+import org.openrepose.filters.openstackidentityv3.OpenStackIdentityV3Filter.FilterConfiguration
 import org.openrepose.filters.openstackidentityv3.objects._
 import org.openrepose.filters.openstackidentityv3.utilities._
 import org.springframework.http.HttpHeaders
 import spray.json._
 
-import scala.collection.JavaConverters._
+import scala.collection.JavaConversions._
 import scala.util.matching.Regex
 import scala.util.{Failure, Success, Try}
 
-class OpenStackIdentityV3Handler(identityConfig: OpenstackIdentityV3Config, identityAPI: OpenStackIdentityV3API)
-  extends AbstractFilterLogicHandler with HttpDelegationManager with LazyLogging {
-
-  private val identityServiceUri = identityConfig.getOpenstackIdentityService.getUri
-  private val forwardGroups = identityConfig.isForwardGroups
-  private val forwardCatalog = identityConfig.isForwardCatalog
-  private val delegatingWithQuality = Option(identityConfig.getDelegating).map(_.getQuality)
-  private val projectIdUriRegex = Option(identityConfig.getValidateProjectIdInUri).map(_.getRegex.r)
-  private val projectIdPrefixes = Try(identityConfig.getValidateProjectIdInUri.getStripTokenProjectPrefixes.split('/')).getOrElse(Array.empty[String])
-  private val bypassProjectIdCheckRoles = Option(identityConfig.getRolesWhichBypassProjectIdCheck).map(_.getRole.asScala.toList)
-  private val configuredServiceEndpoint = Option(identityConfig.getServiceEndpoint) map { serviceEndpoint =>
-    Endpoint(id = "configured-endpoint",
-      url = serviceEndpoint.getUrl,
-      name = Option(serviceEndpoint.getName),
-      interface = Option(serviceEndpoint.getInterface),
-      region = Option(serviceEndpoint.getRegion))
-  }
+class OpenStackIdentityV3Handler(filterConfig: FilterConfiguration,
+                                 identityApi: OpenStackIdentityV3API)
+  extends HttpDelegationManager with LazyLogging {
+
+  import OpenStackIdentityV3Handler._
+
+  def handleRequest(wrappedHttpRequest: HttpServletRequestWrapper, httpResponse: HttpServletResponse): Option[FilterFailure] = {
+    // Track whether or not a failure has occurred so that we can stop checking the request after we know it is bad
+    var filterFailure: Option[FilterFailure] = None
+
+    // Extract the tracing GUID from the request
+    val tracingHeader = Option(wrappedHttpRequest.getHeader(CommonHttpHeader.TRACE_GUID.toString))
+
+    // Attempt to validate the request token with the Identity service
+    val token = authenticate(wrappedHttpRequest, tracingHeader) match {
+      case Success(tokenObject) =>
+        Some(tokenObject)
+      case Failure(e: InvalidSubjectTokenException) =>
+        filterFailure = Some(FilterFailure(HttpServletResponse.SC_UNAUTHORIZED, e.getMessage))
+        None
+      case Failure(e: IdentityServiceOverLimitException) =>
+        filterFailure = Some(FilterFailure(HttpServletResponse.SC_SERVICE_UNAVAILABLE, e.getMessage))
+        authServiceOverLimit(e, httpResponse)
```

Comment:
Will this impact delegation?  Previously this was only called if delegation was turned off.